plugins/inputs/conntrack/conntrack.go

```diff
@@ -57,9 +62,32 @@ func (*Conntrack) SampleConfig() string {
 	return sampleConfig
 }
 
-func (c *Conntrack) Gather(acc telegraf.Accumulator) error {
+func (c *Conntrack) Init() error {
 	c.setDefaults()
+	collectConfig := []string{"all", "percpu"}
+	if err := choice.CheckSlice(c.Collect, collectConfig); err != nil {
+		return fmt.Errorf("config option 'collect': %w", err)
+	}
+
+	for _, collect := range c.Collect {
+		if collect == "all" {
+			all = true
+		}
+		if collect == "percpu" {
+			perCPU = true
+		}
+	}
```

Comment:
Another option and I think I prefer this is to use a switch statement while looping through the c.Collect value and the default case can throw an error on an unrecognized option.

Reply:
Good idea, that will be easier to read.